kyo-core/shared/src/main/scala/kyo/Result.scala

```diff
@@ -65,69 +60,69 @@ object Result:
 
         inline def isFailure: Boolean = !isSuccess
 
-        inline def get: T =
+        inline def fold[U](inline ifFailure: Throwable => U)(inline ifSuccess: T => U): U =
             (self: @unchecked) match
-                case Success(value)     => value
-                case Failure(exception) => throw exception
+                case self: Failure[T] =>
+                    ifFailure(self.exception)
+                case _ =>
+                    try
+                        self match
+                            case self: SuccessFailure[T] @unchecked =>
+                                ifSuccess(self.unnest.asInstanceOf[T])
+                            case self =>
+                                ifSuccess(self.asInstanceOf[T])
+                    catch
+                        case ex if NonFatal(ex) => ifFailure(ex)
+
+        inline def get: T =
+            fold(ex => throw ex)(v => v)
 
         inline def getOrElse[U >: T](inline default: => U): U =
-            (self: @unchecked) match
-                case Success(value) => value
-                case Failure(_)     => default
+            fold(_ => default)(v => v)
 
-        inline def orElse[U >: T](inline alternative: => Result[U]): Result[U] =
-            (self: @unchecked) match
-                case Success(value) => self
-                case Failure(_)     => alternative
+        def orElse[U >: T](alternative: => Result[U]): Result[U] =
```

Comment:
Not sure why I started getting compilation errors for this method and had to remove `inline`.